report_py3o/models/ir_actions_report.py

```diff
@@ -70,6 +77,77 @@ def _get_py3o_filetypes(self):
         "by default Odoo will generate a ZIP file that contains as many "
         "files as selected records. If you enable this option, Odoo will "
         "generate instead a single report for the selected records.")
+    lo_bin_path = fields.Char(
+        string="Path to the libreoffice runtime",
+        compute="_compute_lo_bin_path"
+        )
+    is_py3o_report_not_available = fields.Boolean(
+        compute='_compute_py3o_report_not_available'
+        )
+    msg_py3o_report_not_available = fields.Char(
+        compute='_compute_py3o_report_not_available'
+        )
+
+    @api.model
+    def _register_hook(self):
+        self._validate_reports()
+
+    @api.model
+    def _validate_reports(self):
+        """Check if the existing py3o reports should work with the current
+        installation.
+
+        This method log a warning message into the logs for each report
+        that should not work.
+        """
+        for report in self.search([("report_type", "=", "py3o")]):
+            if report.is_py3o_report_not_available:
+                logger.warning(report.msg_py3o_report_not_available)
+
+    @api.model
+    def _get_lo_bin(self):
+        lo_bin = self.env['ir.config_parameter'].get_param(
```

Comment:
@lmignon a sudo is needed here. cc/ @sbejaoui

Reply:
Good catch @sbidoul!
fixed